From e9cc2b7d184e6bc7aa2d02c66af2db02b44d5316 Mon Sep 17 00:00:00 2001
From: Jasper Orschulko <jasper@fancydomain.eu>
Date: Sun, 25 Jun 2023 23:34:05 +0200
Subject: [PATCH 3/3] python3-platformdirs: Fix recipe version 3.6.0

python3-platformdirs version 3.6.0 requires hatchling build system.
Before this fix it would previously build without issue but throw error when
used in the final image.

Additionally, fixing the ptest which now requires two additional
dependencies.

Upstream-Status: Submitted [https://lists.openembedded.org/g/openembedded-devel/message/103583]

Signed-off-by: Jasper Orschulko <jasper@fancydomain.eu>
---
 .../python/python3-platformdirs_3.6.0.bb       | 18 +++++++-----------
 1 file changed, 7 insertions(+), 11 deletions(-)

diff --git a/meta-python/recipes-devtools/python/python3-platformdirs_3.6.0.bb b/meta-python/recipes-devtools/python/python3-platformdirs_3.6.0.bb
index 20b3299ac7..6444fde7ea 100644
--- a/meta-python/recipes-devtools/python/python3-platformdirs_3.6.0.bb
+++ b/meta-python/recipes-devtools/python/python3-platformdirs_3.6.0.bb
@@ -3,32 +3,28 @@ HOMEPAGE = "https://github.com/platformdirs/platformdirs"
 LICENSE = "BSD-3-Clause"
 LIC_FILES_CHKSUM = "file://LICENSE;md5=ea4f5a41454746a9ed111e3d8723d17a"
 
-SRC_URI = " \
-    git://github.com/platformdirs/platformdirs.git;branch=main;protocol=https \
+SRC_URI += " \
     file://run-ptest \
 "
 
-SRCREV ?= "85b0b67eb33a835b797c1d52dab4a06c5554b7ee"
-SRC_URI[sha256sum] = "7954a68d0ba23558d753f73437c55f89027cf8f5108c19844d4b82e5af396335"
+SRC_URI[sha256sum] = "57e28820ca8094678b807ff529196506d7a21e17156cb1cddb3e74cebce54640"
 
-inherit python_setuptools_build_meta ptest
+inherit pypi python_hatchling ptest
 
 DEPENDS += " \
-    python3-hatch-vcs-native \
-    python3-setuptools-scm-native \
-    python3-toml-native \
+    ${PYTHON_PN}-hatch-vcs-native \
 "
 
-S = "${WORKDIR}/git"
-
 RDEPENDS:${PN}-ptest += " \
     ${PYTHON_PN}-appdirs \
+    ${PYTHON_PN}-covdefaults \
     ${PYTHON_PN}-pytest \
     ${PYTHON_PN}-pytest-mock \
+    ${PYTHON_PN}-pytest-cov \
 "
 
 do_install_ptest() {
-        install -d ${D}${PTEST_PATH}/Tests
+        install -d ${D}${PTEST_PATH}
         cp -rf ${S}/tests ${D}${PTEST_PATH}/
 }
 
-- 
2.41.0

